### PR TITLE
Implement the RS01 consent content

### LINKED ISSUE #527: Implement consent notice for Rally Study 01
This study collects more data than princeton's. We have a unique opportunity to expand on our current study consent experience a bit better.

cc @changecourse

### LINKED ISSUE #530: Add tags to the core add on for the Rally Internal Study 01
Please update these tags to be present in the core add-on. 

card

Community Insights: #D9BFFF 
Product Discovery: #FFB4DC 

**Tags**
Product Discovery,  Rally Community Insights

### LINKED ISSUE #537: Add more context to Demographic form
Add to paragraph at top of demographic form, "Collaborators can join this data to study data to enrich their findings and answer research questions."

### PR DESCRIPTION
**Pull request by hamilton**
Closes #527, closes #530, closes #537

This PR:

1. implements the consent component for RS01 (run `npm run storybook` and check the flow story by clicking on the "join study" button)
2. creates a small switching mechanism to key b/t the princeton irb, the rs01 consent, and a generic consent component (see the same story as 1.). This is also verifiable in the storybook
3. renames the fields for the StudyCard component, and expects them from RS
4. adds a small `parseDateIfNeeded` function in the frontend as a workaround since we don't parse RS dates in the bg script

### CODE REVIEW

File: .storybook/main.cjs [hunk 1]
```diff
@@ -2,7 +2,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-export default {
+module.exports = {
```
Comment (hamilton) [inline .storybook/main.cjs:5]: this was required to make storybook work again.

File: src/routes/demographics/Content.svelte [hunk 2]
```diff
@@ -115,7 +115,9 @@
 
   <slot name="description">
     <p>
-      Each question is completely optional, and can be updated at any time by clicking Manage Profile. The answers you give will help us understand the composition and representivity of the Rally community.
+      Each question is completely optional, and can be updated at any time by clicking Manage Profile. 
+      The answers you give will help us understand the composition and representivity of the Rally community.
+      Additionally, collaborators will combine your answers with the data collected in the studies you join to enrich their findings and answer research questions.
```
Comment (hamilton) [inline src/routes/demographics/Content.svelte:120]: cc @magorlick @changecourse – please validate this is what you want for the demo change copy.

File: src/routes/current-studies/Content.svelte [hunk 3]
```diff
@@ -31,6 +20,22 @@ function leaveStudy(studyID) {
     dispatch("leave-study", studyID);
 }
 
+function parseDateIfNeeded(date) {
```
Comment (rhelmer) [inline src/routes/current-studies/Content.svelte:23]: JS date parsing is terrible :)

Comment (hamilton) [reply src/routes/current-studies/Content.svelte:23]: indeed :) hope this work around suffices for now. If we maintain a core add-on I want to integrate ts here.

**Review by rhelmer — APPROVED** [review]
OK I've been testing this locally and lgtm, I tried dropping in https://github.com/mozilla-rally/study-remote-settings/pull/1 to `locally-available-studies` (with the URL modified to point to the latest RS01 on AMO) and it seems to be working fine.

**Pull request merged**